### PR TITLE
fix(ce-code-review): replace resolve-base.sh with prose-driven base detection

### LINKED ISSUE #811: ce-code-review resolve-base helper path resolves against project root
Howdy folks! Thanks for sharing this plugin and skills with the world. Human here steering my agent to, hopefully, be helpful. 😇 I noticed a small bug here related to #764 for `ce-code-review`. I thought my clanker did a better job articulating this below so I'll defer the details:

## Problem

The `ce-code-review` skill bundles a helper at:

```text
plugins/compound-engineering/skills/ce-code-review/scripts/resolve-base.sh
```

But `SKILL.md` instructs agents to invoke it as:

```bash
bash scripts/resolve-base.sh
```

Because runtime shell commands execute from the user's project checkout, that relative path resolves against the project root, not the skill/plugin directory. In a normal project without `scripts/resolve-base.sh`, the review can fail or report that the repo is missing the helper even though the helper is bundled with the skill.

## Reproduction

1. Install/use `compound-engineering` v3.7.3.
2. In a project that does not have its own `scripts/resolve-base.sh`, run `/ce-code-review` on the current branch without an explicit `base:` argument.
3. The skill attempts to run `bash scripts/resolve-base.sh` from the project root.
4. The helper is not found, or the agent reports that the repo does not have the expected helper.

## Expected behavior

`ce-code-review` should invoke the bundled helper using a path that resolves to the skill directory, not the user's project root.

For example, this likely needs the same pattern used to fix the related `ce-worktree` issue:

```bash
bash "${CLAUDE_SKILL_DIR:-.}/scripts/resolve-base.sh"
```

or the Codex-compatible equivalent for resolving the skill's own base directory.

## Evidence

In `compound-engineering-v3.7.3`, `plugins/compound-engineering/skills/ce-code-review/SKILL.md` still contains two `RESOLVE_OUT=$(bash scripts/resolve-base.sh)` call sites.

This is the same class of bug as #764, which was fixed for `ce-worktree` by resolving the script path against the skill directory instead of the user CWD.

## Workaround

Passing an explicit base ref avoids this path:

```bash
/ce-code-review base:origin/main
```

but the default standalone review path still appears broken for projects that do not happen to contain a same-named helper.

### PR DESCRIPTION
**Pull request by tmchow**
## Summary

Fixes #811 by removing the failing helper rather than fixing how it's called. Supersedes #812.

`ce-code-review` branch mode and standalone mode invoked `bash scripts/resolve-base.sh` via a bare relative path. At runtime the Bash tool's CWD is the user's project, not the skill directory, so the helper has been silently failing on every invocation — the agent improvised its way through base detection on its own judgment, which is why nobody noticed.

## What changed

- **Delete `scripts/resolve-base.sh`** (and its 300-line test file). The helper's most distinctive feature (fork-safe PR-base resolution) wasn't realized at the call sites that actually used it — PR mode has its own inline fork-safe block, and branch/standalone modes never had a real PR-URL to resolve against.
- **Replace both call sites with prose** that names the priority chain explicitly: PR metadata for the current branch (reusing PR mode's `PR_BASE_REMOTE` block for fork safety) → `origin/HEAD` → `gh repo view` → common branch names. Compute the merge-base, unshallow on shallow clones, stop hard if nothing resolves.
- **Preserve the load-bearing safety guard.** Both modes still explicitly say "do not fall back to `git diff HEAD`" — a review without a base would only show uncommitted changes and silently miss all committed work. The contract test in \`tests/review-skill-contract.test.ts\` now pins this guard phrase per-mode.
- **Drop the `${CLAUDE_SKILL_DIR}` workaround and `allowed-tools` grant** that #812 was going to add — both become unnecessary when there's no bundled script to invoke.

Net: -429 lines (script + script tests), +13 lines (prose + new guard test).

## Why this shape instead of #812

#812 was a correct minimal fix that kept the helper and corrected the call. Discussion in #812 and our review session converged on a different question: does the helper earn its complexity?

Empirically no — users have been hitting "couldn't find resolve-base.sh" errors, the spec said to stop, but the agent improvised and the review still completed with reasonable scope. That's a signal that the deterministic helper wasn't load-bearing in branch/standalone modes. PR mode (which DOES need fork-safe resolution) has its own inline block and is untouched.

The change trades:
- One bash tool call (helper) → two or three (PR check + remote fetch + merge-base) per invocation
- Deterministic script behavior (untested for fork edge cases anyway) → prose-driven with explicit priority chain
- A 4-line `allowed-tools` + `\${CLAUDE_SKILL_DIR}` workaround → nothing

## Test plan

- [x] \`bun test\` (1340 pass, 0 fail — net -9 tests = the deleted resolve-base-script.test.ts)
- [x] \`bun run release:validate\` (in sync, 49 agents / 37 skills)
- [x] Contract test \`fails closed when merge-base is unresolved\` still pins the no-HEAD-fallback guard via the new "Do not fall back to \`git diff HEAD\`" regex (matches once per mode = 2 occurrences)
- [ ] Manual: run \`/ce-code-review\` standalone on a clean feature branch in this repo and verify base detection lands on \`origin/main\`
- [ ] Manual: run \`/ce-code-review some-branch\` with no PR open and verify origin/HEAD path
- [ ] Manual: run \`/ce-code-review <PR-URL>\` and verify PR mode still works unchanged

## Credit

Thanks to @frankstallone for the original report (#811) and PR (#812). The closing comment on #812 explains the structural choice.

### CODE REVIEW

File: plugins/compound-engineering/skills/ce-code-review/SKILL.md [hunk 1]
```diff
@@ -292,15 +292,13 @@ If the output is non-empty, inform the user: "You have uncommitted changes on th
...
-if [ -z "$RESOLVE_OUT" ] || echo "$RESOLVE_OUT" | grep -q '^ERROR:'; then echo "${RESOLVE_OUT:-ERROR: resolve-base.sh produced no output}"; exit 1; fi
-BASE=$(echo "$RESOLVE_OUT" | sed 's/^BASE://')
-```
+**If a PR exists for `<branch>`** (check with `gh pr view <branch> --json baseRefName,url`): reuse PR mode's `PR_BASE_REMOTE` block above. Use `baseRefName` as `<base>` and derive `<base-repo>` from the PR URL (e.g., `EveryInc/foo` from `https://github.com/EveryInc/foo/pull/123`). The block already sets `$BASE` to the merge-base SHA — `origin` may point at the user's fork, which is why naive `origin/<base>` is unsafe and the fork-safe block is required.
+
+**If no PR exists**: derive the default branch. Primary source is `git symbolic-ref --quiet --short refs/remotes/origin/HEAD | sed 's#^origin/##'`; fall back to `gh repo view --json defaultBranchRef --jq '.defaultBranchRef.name'`, then to the first of `main`/`master`/`develop`/`trunk` that exists as `origin/<name>` locally. Compute `BASE=$(git merge-base HEAD origin/<base-branch>)`. If `BASE` is empty and the clone is shallow (`git rev-parse --is-shallow-repository`), run `git fetch --unshallow origin` and retry.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply plugins/compound-engineering/skills/ce-code-review/SKILL.md]: > The new no-PR base resolution path only computes `BASE` from `origin/<base-branch>` and retries with `git fetch --unshallow origin`, but it no longer falls back to a local base ref (for example `main`) when the remote-tracking ref is missing.

Addressed: extended the prose to accept bare local refs in two places — the default-branch search now matches `origin/<name>` *or* bare `<name>` locally, and the merge-base step computes against `origin/<base-branch>` when available, otherwise the bare local `<base-branch>`. Single-branch clones, missing-origin repos, and unfetched defaults no longer hard-stop.
